primitives/state-machine/src/overlayed_changes/mod.rs

```diff
@@ -518,12 +547,12 @@ impl OverlayedChanges {
 			.expect("Changes trie transaction was generated by `changes_trie_root`; qed");
 
 		let (main_storage_changes, child_storage_changes) = self.drain_committed();
+		let offchain_storage_changes = self.offchain_drain_committed().collect();
 
 		Ok(StorageChanges {
 			main_storage_changes: main_storage_changes.collect(),
 			child_storage_changes: child_storage_changes.map(|(sk, it)| (sk, it.0.collect())).collect(),
-			#[cfg(feature = "std")]
-			offchain_storage_changes: Default::default(),
+			offchain_storage_changes,
```

Comment:
I am wondering how the changes were communicated from state-machine to client-db, with this value set to default?

Reply:
cc @bkchr , I am not hundred percent sure things were not passing through the &mut parameter (something I did remove in this PR (see macro change)), but yes client db probably don't read from the right data.

Reply:
Which macro changes?

Reply:
Why wasn't it reading from the correct data?

Reply:
That's the part you also change in #7940 (just terrible way of saying it :(

Reply:
TLDR, everything is okay here?

Reply:
Yes :)

Reply:
Perfect :D Then merge it :D